lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3303,21 +3310,28 @@ class ConvertAtenOnesOp : public OpConversionPattern<AtenOnesOp> {
       sizeIndex.push_back(castIntToIndex(rewriter, loc, size[i]));
 
     RankedTensorType newResultType =
-        getTypeConverter()->convertType(op.getType()).cast<RankedTensorType>();
+        getTypeConverter()
+            ->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
     Type outElementType = newResultType.getElementType();
 
-    Value one = rewriter.create<arith::ConstantOp>(
+    int64_t elementValue = 0;
```

Comment:
Can be replaced by `int64_t elementVal = isa<AtenOnesOp>(op) ? 1 : 0`.

Reply:
This will not work because the value returned by `isa<AtenOnesOp>(op)` is not of the bool type.